packages/shylu/shylu_dd/bddc/example/example1.C

```diff
@@ -0,0 +1,210 @@
+
+//@HEADER
+// ************************************************************************
+// 
+//               ShyLU: Hybrid preconditioner package
+//                 Copyright 2012 Sandia Corporation
+// 
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov) 
+// 
+// ************************************************************************
+//@HEADER
+
+#include <mpi.h>
+#include "ShyLU_DDBDDC_config.h"
+#if defined(HAVE_SHYLU_DDBDDC_SHYLU_NODETACHO)
+#include "Kokkos_Core.hpp"
+#endif
+#include "shylu_PreconditionerBDDC.h"
+#include "shylu_KrylovSolver.h"
+#include "setupTest.h"
+#include <Teuchos_XMLParameterListHelpers.hpp>
+
+using Teuchos::RCP;
+
+typedef int LO; // Local Ordinal
+typedef long GO; // Global Ordinal
```

Comment:
Followed your suggestion and am now using Tpetra's default global ordinal type.

Reply:
Here is how you can ensure that you always get a 64-bit global ordinal type:
```c++
// Your application prefers long, though I don't 
// necessarily agree that this is a good idea. 
#if defined(HAVE_TPETRA_INST_INT_LONG)
  using GO = long;
#elif defined(HAVE_TPETRA_INST_INT_LONG_LONG)
  using GO = long long;
#elif defined(HAVE_TPETRA_INST_INT_UNSIGNED_LONG)
  using GO = unsigned long;
#  error "Tpetra does not enable one of the possibly 64-bit global ordinal types."
#endif
static_assert (sizeof (GO) >= size_t (8), "GO is not a 64-bit type.");
```

Reply:
Once Tpetra finishes deprecation and removal of the GlobalOrdinal template parameter, you should be able to remove the macros and just have one typedef for `GO` and a `static_assert`.  The typedef for GO would look like this: `using GO = Tpetra::Map::global_ordinal_type;`.  Right now, since the template parameters are still there, you could write `using GO = Tpetra::Map<>::global_ordinal_type;`.